src/main/java/pl/bartoszmech/infrastructure/loginandregister/controller/RegisterController.java

```diff
@@ -0,0 +1,31 @@
+package pl.bartoszmech.infrastructure.loginandregister.controller;
+
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import pl.bartoszmech.domain.accountidentifier.AccountIdentifierFacade;
+import pl.bartoszmech.domain.accountidentifier.dto.RegistrationResultDto;
+import pl.bartoszmech.domain.accountidentifier.dto.UserDto;
+import pl.bartoszmech.infrastructure.loginandregister.controller.dto.RegisterRequestDto;
+
+import static org.springframework.http.HttpStatus.CREATED;
+
+@RestController
+@AllArgsConstructor
```

Comment:
in order to make this controller more elastic, you could add a path at the begining like the name of the controller /accounts.

POST /accounts/register

Or simply 

POST /accounts